### ISSUE TITLE
constDecl need a initializer

### ISSUE THREAD

**Issue by enochii** [antlr/grammars-v4#2071]
Go's const declaration need a initializer. For example:
```go
const a = 1 // ok
const b // not ok
```
possible compiler output maybe like "missing value in const declaration". And in [Parser.g4](https://github.com/antlr/grammars-v4/blob/master/golang/GoParser.g4#L74) things go like this:
```g4
constDecl
    : 'const' (constSpec | '(' (constSpec eos)* ')')
    ;
constSpec
    : identifierList (type_? '=' expressionList)?
    ;
```
So missing value error is permitted and won't be caught by parsing. Though we can do this in later pass(e.g. semantic analysis), I think we could fix this by simply changing `constSpec` to:
```go
constSpec
    : identifierList type_? '=' expressionList
    ;
```
If you think it's OK, I can help to do this.

**Comment by studentmain** [comment]
```go
type Season uint8
const (
	Spring = Season(iota)
	Summer
	Autumn
	Winner
)
```

**Comment by kaby76** [comment]
See the code [here](https://github.com/golang/go/blob/18bd7aa62581f313c86164d763b1e246307888a9/src/cmd/compile/internal/syntax/parser.go#L549). Grammars should be scraped via tool from the implementation.

**Comment by enochii** [comment]
> See the code [here](https://github.com/golang/go/blob/18bd7aa62581f313c86164d763b1e246307888a9/src/cmd/compile/internal/syntax/parser.go#L549). Grammars should be scraped via tool from the implementation, and probably can because most follow the usual pattern of a recursive descent parser. But, the static semantics likely cannot, and should be codified in a spec.

Thanks, I kind of got it. I guess the main reason here should be keeping the parser simple, am I right?

**Comment by kaby76** [comment]
@enochii I think the CFG portion should be faithful to the implementation. But, if you think the parse needs to be influenced by static semantics, then it should be added in a lexer or parser base class like it is done for other grammars in this repo, or a post-parse tree walker implemented.

I don't know enough about Golang to know what the rule is for an initializer. Perhaps you can codify the static semantics rule you are expecting in natural language. Or, point to what the documentation says about initializers for Go. Or, worse case, provide some examples of what the compiler outputs when the initializer is missing. That's the first thing to do here.

**Comment by kaby76** [comment]
And, I should note that many of the grammars in this repo assume a rolling version. So, csharp/, c/, cpp/, etc, are for who-knows-what-version, maybe the latest? The grammar should be clearly labeled per-version of the language.

**Comment by studentmain** [comment]
> So, csharp/, c/, cpp/, etc, are for who-knows-what-version, maybe the latest?

C# 6, C 11, C++ 14, they mentioned it in comment and/or readme. At least we know the lower bound, but it seems not support latest at all.

**Comment by kaby76** [comment]
Yes, I've read in a few places (SO, Reddit, YCombinator,...) comments that the grammars aren't up to snuff, missing a few versions here and there. Something to fix later. But first, the CI should be nailed down (I'm resurrecting StringTemplates for C# from the dead, bringing it up to date).

**Comment by teverett** [comment]
Grammars-v4 was started in 2012, as a result of the Antlrv3 grammars not having been updated as languages and ANTLR 
versions changed.  Those grammars are here:

https://github.com/antlr/grammars-v3

and came from:

https://www.antlr3.org/grammar/list.html

In the past 8 years, we created the java test plugin and integrated CI/CD with the intent that while not every grammar may be complete or an accurate description of the language, that it at least compile with ANTLR and pass some unit tests.   Over the course of the years many grammars have been added, test features added, and there has been more than one CI/CD provider.    

In the last year, there has been a resurgence of interest in the grammars and I now see more PR's in a week than I did in the month.  There is interest in validating the grammars for languages other than Java, a direction I fully support.

Thanks for all your contributions.  It is contributions such as you all are making that make the open-source community the success that it is.

**Comment by kaby76** [comment]
This repo is really a diamond in the rough, but can become absolutely wonderful! Antlr is a wonderful way of capturing the static semantics for languages. I am really psyched to help out.

**Comment by Marti2203** [comment]
Hi @enochii , I think the change you propose should not be added as I have updated the grammar to follow the specification from the 20-th of February, I think such a check should be in semantic analysis. as the first comment you received shows an example where initialization is not needed.
